kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_lagrange_multiplier.h

```diff
@@ -0,0 +1,1184 @@
+                mConstraintFactor = constraint_scale_factor;
+            }
+
+            // Check T has been computed
+            if (TSparseSpace::Size1(BaseType::mT) != BaseType::mSlaveIds.size() || TSparseSpace::Size2(BaseType::mT) != BaseType::mEquationSystemSize) {
+                BaseType::mT.resize(BaseType::mSlaveIds.size(), BaseType::mEquationSystemSize, false);
+                ConstructMasterSlaveConstraintsStructure(rModelPart);
+            }
+
+            // If not previously computed we compute
+            if (BaseType::mOptions.IsNot(TRANSFORMATION_MATRIX_COMPUTED)) {
+                BuildMasterSlaveConstraints(rModelPart);
+            }
+
+            // Extend with the LM constribution
+            const SizeType number_of_slave_dofs = TSparseSpace::Size1(BaseType::mT);
+
+            // Definition of the total size of the system
+            const SizeType total_size_of_system = BaseType::mEquationSystemSize + (BaseType::mOptions.Is(DOUBLE_LAGRANGE_MULTIPLIER) ? 2 * number_of_slave_dofs : number_of_slave_dofs);
+            TSystemVectorType b_modified(total_size_of_system);
+
+            // Copy the RHS
+            #pragma omp parallel for
+            for (int i = 0; i < static_cast<int>(BaseType::mEquationSystemSize); ++i) {
+                b_modified[i] = rb[i];
+            }
+            // Fill with zeros
+            #pragma omp parallel for
+            for (int i = static_cast<int>(BaseType::mEquationSystemSize); i < static_cast<int>(total_size_of_system); ++i) {
+                b_modified[i] = 0.0;
+            }
+            rb.resize(total_size_of_system, false);
+
+            // Compute LM contributions
+            TSystemVectorType b_lm(total_size_of_system);
+            ComputeRHSLMContributions(b_lm, mConstraintFactor);
+
+            // Fill auxiliar vector
+            TSparseSpace::UnaliasedAdd(b_modified, 1.0, b_lm);
+
+            // Finally reassign
+            TSparseSpace::Copy(b_modified, rb);
+        }
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * @brief Applies the constraints with master-slave relation matrix
+     * @param pScheme The integration scheme considered
+     * @param rModelPart The model part of the problem to solve
+     * @param rA The LHS matrix
+     * @param rb The RHS vector
+     */
+    void ApplyConstraints(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemMatrixType& rA,
+        TSystemVectorType& rb
+        ) override
+    {
+        KRATOS_TRY
+
+        if (rModelPart.MasterSlaveConstraints().size() != 0) {
+            // Getting process info
+            const ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+
+            // First build the relation matrix
+            BuildMasterSlaveConstraints(rModelPart);
+
+            // Copy the LHS to avoid memory errors
+            TSystemMatrixType copy_of_A(rA);
```

Comment:
Can't you avoid making those copies. These copies consume a lot of memory as the number of DOFs increase. 
In this function all of A, b and T have a copy. This momentarily will cause memory spike which could be a problem.

Reply:
It cannot be avoided, the rA is a reference that we are going to expand into the new form. In order to build a sparse matrix, you need to build the structure, so I cannot preserve the old values, I must copy them

Reply:
In any case, you don't see it, but the block builder and solver does this, but it does using the SparseMultiplicationUtilities, so you cannot see it directly

Reply:
For the constraints BTW

Reply:
Well all I am saying this will cause memory spikes .. and using this will be restrictive in terms of the size of the problem very quickly. 
Imagine having a A of size x GB (without constraints) , when using this, even if there a very few constraints, (not considering T) one will have a memory of minimum 2x GB of memory for a short while and in the case it does not fit in the memory this is going to cause a crash.

Reply:
I understand, but there is no way to avoid this IMO due to the rigid behaviour of the sparse matrix design

Reply:
@adityaghantasala I found a way to avoid the copy, create an empty sparse matrix and do a swap between them

Reply:
Problem solved